packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -312,45 +314,36 @@ export class TdsDropdown {
   }
 
   private setDefaultOption = () => {
-    // Ensure this.host.children is not empty
-    if (this.host.children.length === 0) {
-      console.warn('TDS DROPDOWN: Data missing. Disregard if loading data asynchronously.');
-      return;
-    }
-
     if (this.defaultValue) {
       const children = Array.from(this.host.children).filter(
         (element) => element.tagName === 'TDS-DROPDOWN-OPTION',
       );
       let matched = false;
 
-      children.forEach((element: HTMLTdsDropdownOptionElement) => {
-        if (this.multiselect) {
-          this.defaultValue.split(',').forEach((value) => {
-            if (value === element.value) {
-              element.setSelected(true);
-              this.value = this.value ? [...this.value, element.value] : [element.value];
-              matched = true;
-            }
-          });
-        } else {
-          if (this.defaultValue === element.value) {
+      const defaultValues = this.multiselect ? this.defaultValue.split(',') : [this.defaultValue];
+
+      // Ensure unique default values
+      const uniqueDefaultValues = [...new Set(defaultValues)];
```

Comment:
```suggestion
const defaultValues = this.multiselect ? new Set(this.defaultValue.split(',')) : new Set([this.defaultValue]);
```

Might be worth looking through this implementation and see if we are instatiating arrays that should be Sets anywhere else